entity-framework/core/providers/in-memory/index.md

```diff
@@ -7,7 +7,10 @@ uid: core/providers/in-memory/index
 ---
 # EF Core In-Memory Database Provider
 
-This database provider allows Entity Framework Core to be used with an in-memory database. The in-memory database can be useful for testing, although the SQLite provider in in-memory mode may be a more appropriate test replacement for relational databases. The in-memory database is designed for testing only. The provider is maintained as part of the [Entity Framework Core Project](https://github.com/dotnet/efcore).
+This database provider allows Entity Framework Core to be used with an in-memory database. While some users use the in-memory database for testing, this is generally discouraged; the SQLite provider in in-memory mode is a more appropriate test replacement for relational databases. For more information on how to test EF Core applications, see the [testing documentation](xref:core/testing/index). The provider is maintained as part of the [Entity Framework Core Project](https://github.com/dotnet/efcore).
+
+> [!WARNING]
+> The In-Memory provider was not designed for production use and should never be used as such.
```

Comment:
I wonder if "production use" is clear here. Maybe, "The In-Memory provider was not designed for use outside of testing environments and should never be used as such?"

Reply:
Better